ggshield/output/output_handler.py

```diff
@@ -13,15 +16,41 @@ def __init__(
         show_secrets: bool,
         verbose: bool,
         output: Optional[str] = None,
-        *args: Any,
-        **kwargs: Any,
     ):
         self.show_secrets = show_secrets
         self.verbose = verbose
         self.output = output
 
-    def process_scan(self, scan: ScanCollection, top: bool = True) -> Tuple[Any, int]:
-        pass
+    def process_scan(self, scan: ScanCollection) -> int:
+        """Process a scan collection, write the report to :attr:`self.output`
+
+        :param scan: The scan collection to process
+        :return: The exit code
+        """
+        text = self._process_scan_impl(scan)
+        if self.output:
+            with open(self.output, "w+") as f:
+                f.write(text)
+        else:
+            click.echo(text)
+        return OutputHandler._get_exit_code(scan)
+
+    @abstractmethod
+    def _process_scan_impl(self, scan: ScanCollection) -> str:
+        """Implementation of scan processing,
+        called by :meth:`OutputHandler.process_scan`
+
+        Must return a string for the report.
+
+        :param scan: The scan collection to process
+        :return: The content
+        """
+        return ""
```

Comment:
minor: you might raise `NotImplementedError` to emphasize it should be implemented.